applications/fishing-map/src/features/datasets/NewDataset.tsx

```diff
@@ -160,27 +164,72 @@ export type DatasetMetadata = {
   name: string
   description?: string
   type: DatasetTypes.Context
-  properties?: { type?: DatasetCustomTypes }
+  configuration?: {
+    file?: string
+    type?: DatasetCustomTypes
+    format?: 'geojson'
+  }
 }
 
 function NewDataset(): React.ReactElement {
   const { t } = useTranslation()
   const { datasetModal, dispatchDatasetModal } = useDatasetModalConnect()
+  const { addNewDatasetToWorkspace } = useNewDatasetConnect()
 
   const [file, setFile] = useState<File | undefined>()
   const [loading, setLoading] = useState(false)
   const [error, setError] = useState('')
   const [metadata, setMetadata] = useState<DatasetMetadata | undefined>()
+  const locationType = useSelector(selectLocationType)
   const { dispatchCreateDataset } = useDatasetsAPI()
 
-  const onFileLoaded = (file: File) => {
-    setFile(file)
-    setMetadata((metadata) => ({
-      ...metadata,
-      name: capitalize(lowerCase(file.name.split('.')[0])),
-      type: DatasetTypes.Context,
-    }))
-  }
+  const onFileLoaded = useCallback(
+    async (file: File) => {
+      setLoading(true)
+      if (error) {
+        setError('')
+      }
+      const isZip = file.type === 'application/zip'
+      const isGeojson =
+        !isZip && (file.type === 'application/json' || file.name.includes('.geojson'))
+      let geojson: FeatureCollectionWithFilename | undefined = undefined
+      if (isZip) {
+        try {
+          const shpjs = await import('shpjs').then((module) => module.default)
+          const fileData = await blobToArrayBuffer(file)
+          // TODO support multiple files in shapefile
+          geojson = (await shpjs(fileData)) as FeatureCollectionWithFilename
+        } catch (e) {
+          console.warn('Error reading file:', e)
+        }
+      } else {
+        const fileData = await blobToText(file)
+        try {
+          geojson = JSON.parse(fileData)
+        } catch (e) {
+          console.warn('Error reading file:', e)
+        }
+      }
+      if (geojson !== undefined) {
+        setFile(file)
+        setMetadata((metadata) => ({
+          ...metadata,
+          name: capitalize(lowerCase(file.name.split('.')[0])),
```

Comment:
I imported it from lodash and when I saw the bundle size:
![image](https://user-images.githubusercontent.com/10500650/107057825-24058000-67d4-11eb-8e9d-a4e1cd669301.png)
I thought for something as simple as that would be better to use our our version, but maybe it doesn't really change anything. do you prefer lodash?

Reply:
Good catch! I replaced the function with this:
```js
file.name.lastIndexOf('.') > 0 ? file.name.substr(0, file.name.lastIndexOf('.')) : file.name
```
to ensure it works when there is no extension (it shouldn't happen, but just in case) and we remove only the extension for the name